### PR TITLE
Update of HGCal validation of CaloParticle

### PR DESCRIPTION
**Pull request by ebrondol**
#### PR description:

- Extend CaloParticle validation in `HGCalValidator` and `makeHGCalValidationPlots` (including self-energy, energy sum per layer, first and last layer, ...)
- Remove CaloParticle validation in `CaloParticleValidation` (still PFvalidation is in the code, it will be restructured)
- Fix `--separate` option for 2D plots in `makeHGCalValidationPlots` 

#### PR validation:

- Results for 2 single pion files can be found [here](https://ebrondol.web.cern.ch/ebrondol/HGCal/addCaloPartValidation/20201201_vanilla_separate/) for vanilla release and [here](https://ebrondol.web.cern.ch/ebrondol/HGCal/addCaloPartValidation/20201207_fb990d2_separate/) for this branch

### CODE REVIEW

**Comment by ebrondol** [comment]
@rovere @apsallid @felicepantaleo

**Comment by jfernan2** [comment]
please test

File: Validation/HGCalValidation/interface/HGVHistoProducerAlgo.h [hunk 1]
```diff
@@ -155,7 +170,7 @@ class HGVHistoProducerAlgo {
   using Histograms = HGVHistoProducerAlgoHistograms;
 
   void bookInfo(DQMStore::IBooker& ibook, Histograms& histograms);
-  void bookCaloParticleHistos(DQMStore::IBooker& ibook, Histograms& histograms, int pdgid);
+  void bookCaloParticleHistos(DQMStore::IBooker& ibook, Histograms& histograms, int pdgid, unsigned layers);
```
Comment (rovere) [inline Validation/HGCalValidation/interface/HGVHistoProducerAlgo.h:173]: while `unsigned` is valid, maybe `unsigned int` would be preferable?

Comment (ebrondol) [reply Validation/HGCalValidation/interface/HGVHistoProducerAlgo.h]: Ok, changed accordingly everywhere.

**Comment by rovere** [comment]
@cmsbuild please test

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
+1
the differences https://github.com/cms-sw/cmssw/pull/32412#issuecomment-742646143 are compatible with the PR description (plots moved from HGCAL/CaloParticles  to HGCAL/HGCalValidator/SelectedCaloParticles and addition of new plots)

**Pull request merged**